discovery/utils/inventory.py

```diff
@@ -60,11 +62,62 @@ def put_inventory_data(self, data):
 class InventorySanitizer:
 
     @staticmethod
-    def sanatize(inventory_data: dict) -> dict:
-        pass
+    def sanitize(inventory_data: dict) -> dict:
+        InventorySanitizer.typecast(inventory_data)
+        list_groups = ConfluentServices.get_all_group_names()
+        list_aggregator = ['ssl_enabled', 'rbac_enabled', 'rbac_enabled_public_pem_path',
+                           'ssl_keystore_alias', 'ssl_keystore_key_password', 'ssl_keystore_store_password',
+                           'ssl_mutual_auth_enabled', 'ssl_provided_keystore_and_truststore',
+                           'ssl_provided_keystore_and_truststore_remote_src', 'ssl_truststore_ca_cert_alias',
+                           'ssl_truststore_password']
+
+        # check if the given var:value is defined under vars for all component group
+        for item in list_aggregator:
+            value = None
+            aggregate = False
+            for group in list_groups:
+                aggregate = True
+                if group in inventory_data:
```

Comment:
We should be checking only the groups available in the inventory. I feel this can be simplified further.